app/models/topic.rb

```diff
@@ -1050,9 +1052,7 @@ def invite(invited_by, username_or_email, group_ids = nil, custom_message = nil)
         raise NotAllowed.new(I18n.t("topic_invite.muted_topic"))
       end
 
-      if !target_user.staff? &&
-         target_user&.user_option&.enable_allowed_pm_users &&
-         !AllowedPmUser.where(user: target_user, allowed_pm_user: invited_by).exists?
+      if !target_user.staff? && target_user_communication_preferences&.disallowing_pms?
```

Comment:
Design thought: We are often doing these `staff?` checks outside of the defender once the preferences are returned? Should `disallowing_pms?` just be like this:

```ruby
 def disallowing_pms?
  return false if acting_user.staff?
  is_disallowing_all_pms || is_disallowing_pms_from_acting_user
end
```